packages/jest-runtime/src/transform.js

```diff
@@ -110,94 +116,189 @@ const readCacheFile = (filePath: Path, cachePath: Path): ?string => {
   return fileData;
 };
 
-module.exports = (
-  filename: Path,
-  config: ?Config,
-  options: ?TransformOptions,
-): vm.Script => {
+const getScriptCacheKey = (filename, config) => {
   const mtime = fs.statSync(filename).mtime;
-  const instrumentCacheKey = (options && options.instrument ? '1' : '0');
-  const key = filename + '-' + mtime.getTime() + '-' + instrumentCacheKey;
-
-  if (cache.has(key)) {
-    const content = cache.get(key);
-    if (content) {
-      return content;
-    }
-  }
-
-  let content = fs.readFileSync(filename, 'utf8');
-  // If the file data starts with a shebang remove it. Leaves the empty line
-  // to keep stack trace line numbers correct.
-  if (content.startsWith('#!')) {
-    content = content.replace(/^#!.*/, '');
-  }
+  return `${filename}_${mtime.getTime()}_` +
```

Comment:
this is not lame. It seems like template literals are about 4 times slower than concatenation, and this is one of the hottest paths in the codebase
cc @slevithan string performance still matters :)